### PR TITLE
Describe team attribute for set_pipeine step

### PR DESCRIPTION
**Pull request by tlwr**
Adds documentation for https://github.com/concourse/concourse/pull/5729

Aside, the new docs are awesome, as is booklit

### CODE REVIEW

File: lit/docs/jobs/steps.lit [hunk 1]
```diff
@@ -679,6 +679,23 @@ by configuring \reference{schema.step.on_failure} or
...
+
+      \optional-attribute{team}{identifier}{
+        By default the \code{set_pipeline} step sets pipelines for the
+        pipeline's \reference{teams}{team}.
+
+        The \code{team} attribute can be used to specify a specific team.
+
+        Only the \reference{main-team} is allowed to set another team's
+        pipeline.  Any team other than the \reference{main-team} using the
+        \code{team} attribute will error.
```
Comment (tlwr) [inline lit/docs/jobs/steps.lit:691]:
```suggestion
        Only the \reference{main-team} is allowed to set another team's
        pipeline.  Any team other than the \reference{main-team} using the
        \code{team} attribute will error, unless they reference their own team.
```

File: lit/docs/jobs/steps.lit [hunk 2]
```diff
@@ -679,6 +679,23 @@ by configuring \reference{schema.step.on_failure} or
...
+        The \code{team} attribute can be used to specify a specific team.
+
+        Only the \reference{main-team} is allowed to set another team's
+        pipeline.  Any team other than the \reference{main-team} using the
+        \code{team} attribute will error, unless they reference their own team.
+
+        \warn{
+          The \code{team} attribute was introduced in Concourse v6.4.0. It is
+          considered an \bold{experimental} feature until its associated
+          \link{RFC}{https://github.com/concourse/rfcs/pull/31} is resolved.
```
Comment (vito) [inline lit/docs/jobs/steps.lit:696]: Can we add scarier wording and a link to the Discussion?

File: lit/docs/jobs/steps.lit [hunk 3]
```diff
@@ -679,6 +679,23 @@ by configuring \reference{schema.step.on_failure} or
         in multiple files, the value from a file that is passed later in the
         list will override the values from files earlier in the list.
       }
+
+      \optional-attribute{team}{identifier}{
+        By default the \code{set_pipeline} step sets pipelines for the
+        pipeline's \reference{teams}{team}.
```
Comment (vito) [inline lit/docs/jobs/steps.lit:685]: This sounds a little ambiguous; how about something like this?:

```suggestion
        By default, the \code{set_pipeline} step sets the pipeline within the same \reference{teams}{team} that is running the build.
```

File: lit/docs/jobs/steps.lit [hunk 4]
```diff
@@ -679,6 +679,23 @@ by configuring \reference{schema.step.on_failure} or
...
+
+        The \code{team} attribute can be used to specify a specific team.
```
Comment (vito) [inline lit/docs/jobs/steps.lit:687]: I wonder if we could frame this as "the default value for this field is the team that is running the build"?

**Comment by tlwr** [comment]
I think I've incorporated your suggestions in the latest commit

I hope it reads clearer now
image

**Pull request merged**